include/openssl/evp.h

```diff
@@ -691,20 +691,23 @@ OPENSSL_EXPORT int EVP_PKEY_keygen_init(EVP_PKEY_CTX *ctx);
 // containing the result. It returns one on success or zero on error.
 OPENSSL_EXPORT int EVP_PKEY_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY **out_pkey);
 
-OPENSSL_EXPORT int EVP_PKEY_encapsulate_init(EVP_PKEY_CTX *ctx,
-                                             const OSSL_PARAM params[]);
+// EVP_PKEY_encapsulate is an operation defined for a KEM (Key Encapsulation
+// Mechanism). The function generates a shared secret and a ciphertext which
+// encapsulates the shared secret using the public key given within |ctx|.
```

Comment:
TBD word smiting, can this be a bit more clear what is written out vs passed into this function, something like:
```suggestion
// EVP_PKEY_encapsulate is an operation defined for a KEM (Key Encapsulation
// Mechanism). The function generates a shared secret, writes it to |shared_secret|, and
// encapsulates with the public key in |ctx| and written to |ciphertext|.
```

Also can you update it to specify what is allocated vs expected to already be allocated